beetsplug/parentwork.py

```diff
@@ -0,0 +1,200 @@
+
+        composer_exists = False
+        if 'artist-relation-list' in work_info['work']:
+            for artist in work_info['work']['artist-relation-list']:
+                if artist['type'] == 'composer':
+                    parent_composer.append(artist['artist']['name'])
+                    parent_composer_sort.append(artist['artist']['sort-name'])
+
+            parentwork_info['parent_composer'] = u', '.join(parent_composer)
+            parentwork_info['parent_composer_sort'] = u', '.join(
+                    parent_composer_sort)
+
+        if not composer_exists:
+            self._log.debug('no composer for {}; add one at \
+https://musicbrainz.org/work/{}', item, work_info['work']['id'])
+
+        parentwork_info['parentwork'] = work_info['work']['title']
+        parentwork_info['mb_parentworkid'] = work_info['work']['id']
+
+        if 'disambiguation' in work_info['work']:
+            parentwork_info['parentwork_disambig'] = work_info[
+                    'work']['disambiguation']
+
+        else:
+            parentwork_info['parentwork_disambig'] = None
+
+        return parentwork_info
+
+    def find_work(self, item, force):
+        """Finds the parent work of a recording and populates the tags
+        accordingly.
+
+        Namely, the tags parentwork, parentwork_disambig, mb_parentworkid,
+        parent_composer, parent_composer_sort and work_date are populated.
+        """
+
+        if hasattr(item, 'parentwork'):
+            hasparent = True
+        else:
+            hasparent = False
+        if not item.mb_workid:
+            self._log.info('No work for {}, \
+add one at https://musicbrainz.org/recording/{}', item, item.mb_trackid)
+            return
+        if force or (not hasparent):
+            try:
+                work_info, work_date = find_parentwork_info(item.mb_workid)
+            except musicbrainzngs.musicbrainz.WebServiceError as e:
+                self._log.debug("error fetching work: {}", e)
+                return
+            parent_info = self.get_info(item, work_info)
+            if 'parent_composer' in parent_info:
+                self._log.debug("Work fetched: {} - {}",
+                                parent_info['parentwork'],
+                                parent_info['parent_composer'])
+            else:
+                self._log.debug("Work fetched: {} - no parent composer",
+                                parent_info['parentwork'])
+
+        elif hasparent:
+            self._log.debug("{} : Work present, skipping", item)
+            return
+
+        for key, value in parent_info.items():
+            if value:
+                item[key] = value
+
+        if work_date:
+            item['work_date'] = work_date
+        ui.show_model_changes(
+            item, fields=['parentwork', 'parentwork_disambig',
+                          'mb_parentworkid', 'parent_composer',
+                          'parent_composer_sort', 'work_date'])
+
+        item.store()
```

Comment:
Delete this call to `item.store()`; the item is already stored in both places where this function is called.

Reply:
this might actually solve the problem I had with catching database changes, as the error was on the ```item.store()``` . But it's for another time I guess.